tutorials/asr/04_Online_Offline_Speech_Commands_Demo.ipynb

```diff
@@ -0,0 +1,732 @@
+    "import nemo.collections.asr as nemo_asr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# sample rate, Hz\n",
+    "SAMPLE_RATE = 16000"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Restore the model from NGC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "mbn_model = nemo_asr.models.EncDecClassificationModel.from_pretrained(\"MatchboxNet-3x1x64-v2\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since speech commands model MatchBoxNet doesn't consider non-speech scenario, \n",
+    "here we use a vad model to help reduce false alarm for background noise/silence.\n",
+    "Please note the **VAD model is not perfect for various microphone input** and you might need to finetune on your input and play with different parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "vad_model = nemo_asr.models.EncDecClassificationModel.from_pretrained('MatchboxNet-VAD-3x2')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Observing the config of the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from omegaconf import OmegaConf\n",
+    "import copy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Preserve a copy of the full config\n",
+    "vad_cfg = copy.deepcopy(vad_model._cfg)\n",
+    "mbn_cfg = copy.deepcopy(mbn_model._cfg)\n",
+    "print(OmegaConf.to_yaml(mbn_cfg))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What classes can this model recognize?\n",
+    "\n",
+    "Before we begin inference on the actual audio stream, lets look at what are the classes this model was trained to recognize"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "labels = mbn_cfg.labels\n",
+    "for i in range(len(labels)):\n",
+    "    print('%-10s' % (labels[i]), end=' ')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup preprocessor with these settings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set model to inference mode\n",
+    "mbn_model.eval();\n",
+    "vad_model.eval();"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mbn_model = mbn_model.to(mbn_model.device)\n",
```

Comment:
This line is odd - set the device to its own device. Maybe do a torch.cuda.is_available() check, create the appropriate device (CPU/GPU) and pass that device here instead.

Reply:
lol yes. That's was for cpu or to fix container issues (?) I got from vad notebook a while ago. we can just drop these lines and keep 
`audio_signal, audio_signal_len = audio_signal.to(model.device), audio_signal_len.to(model.device)` in infer_signal
let me check it within container.